devito/arch/compiler.py

```diff
@@ -719,9 +732,9 @@ def __lookup_cmds__(self):
         # we try to use `mpiicc` first, while `mpicc` is our fallback, which may
         # or may not be an Intel distribution
         try:
-            check_output(["mpiicc", "--version"]).decode("utf-8")
+            check_output(["mpiicc", "-cc=%s" % self.CC, "--version"]).decode("utf-8")
             self.MPICC = 'mpiicc'
```

Comment:
are we sure the OneAPI MPI compiler name is still mpiicc, rather than say mpiicx ?

Reply:
Yes